File: backend/src/services/email_service.py
"""
Datrix™ Business Intelligence Scanner
Email Service for sending notifications
"""

import os
import smtplib
from email.mime.text import MIMEText
from email.mime.multipart import MIMEMultipart
from typing import Optional

class EmailService:
    """Email service for sending notifications"""
    
    def __init__(self):
        self.smtp_host = os.getenv('SMTP_HOST', 'smtp.gmail.com')
        self.smtp_port = int(os.getenv('SMTP_PORT', '587'))
        self.smtp_user = os.getenv('SMTP_USER', '')
        self.smtp_password = os.getenv('SMTP_PASSWORD', '')
        self.from_email = os.getenv('FROM_EMAIL', self.smtp_user)
        self.from_name = os.getenv('FROM_NAME', 'Datrix™ Business Intelligence')
        
    def send_email(self, to_email: str, subject: str, html_body: str, text_body: str = None) -> bool:
        """Send an email"""
        try:
            if not self.smtp_user or not self.smtp_password:
                print("Email service not configured - skipping email send")
                return False
            
            msg = MIMEMultipart('alternative')
            msg['Subject'] = subject
            msg['From'] = f"{self.from_name} <{self.from_email}>"
            msg['To'] = to_email
            
            # Add text and HTML parts
            if text_body:
                part1 = MIMEText(text_body, 'plain')
                msg.attach(part1)
            
            part2 = MIMEText(html_body, 'html')
            msg.attach(part2)
            
            # Send email
            with smtplib.SMTP(self.smtp_host, self.smtp_port) as server:
                server.starttls()
                server.login(self.smtp_user, self.smtp_password)
                server.send_message(msg)
            
            print(f"✓ Email sent successfully to {to_email}")
            return True
            
        except Exception as e:
            print(f"Failed to send email to {to_email}: {str(e)}")
            return False
    
    def send_verification_email(self, to_email: str, full_name: str, verification_code: str) -> bool:
        """Send verification code email"""
        subject = "Verify Your Datrix™ Account"
        
        html_body = f"""
        <!DOCTYPE html>
        <html>
        <head>
            <style>
                body {{ font-family: Arial, sans-serif; line-height: 1.6; color: #333; }}
                .container {{ max-width: 600px; margin: 0 auto; padding: 20px; }}
                .header {{ background: linear-gradient(135deg, #667eea 0%, #764ba2 100%); color: white; padding: 30px; text-align: center; border-radius: 10px 10px 0 0; }}
                .content {{ background: #f9f9f9; padding: 30px; border-radius: 0 0 10px 10px; }}
                .code-box {{ background: white; border: 2px dashed #667eea; padding: 20px; text-align: center; margin: 20px 0; border-radius: 8px; }}
                .code {{ font-size: 32px; font-weight: bold; color: #667eea; letter-spacing: 5px; }}
                .footer {{ text-align: center; margin-top: 20px; color: #666; font-size: 12px; }}
                .button {{ background: #667eea; color: white; padding: 12px 30px; text-decoration: none; border-radius: 5px; display: inline-block; margin: 20px 0; }}
            </style>
        </head>
        <body>
            <div class="container">
                <div class="header">
                    <h1>Datrix™</h1>
                    <p>Business Intelligence Scanner</p>
                </div>
                <div class="content">
                    <h2>Hello {full_name},</h2>
                    <p>Thank you for registering with Datrix™ Business Intelligence Scanner!</p>
                    <p>Please use the verification code below to complete your registration:</p>
                    
                    <div class="code-box">
                        <div class="code">{verification_code}</div>
                    </div>
                    
                    <p><strong>Important:</strong></p>
                    <ul>
                        <li>This code is valid for 30 minutes</li>
                        <li>Do not share this code with anyone</li>
                        <li>After verification, your account will be reviewed by our admin team</li>
                    </ul>
                    
                    <p>If you didn't request this verification, please ignore this email.</p>
                    
                    <div class="footer">
                        <p>© 2025 Navvi Corporation. All rights reserved.</p>
                        <p>This is an automated email. Please do not reply.</p>
                    </div>
                </div>
            </div>
        </body>
        </html>
        """
        
        text_body = f"""
        Hello {full_name},
        
        Thank you for registering with Datrix™ Business Intelligence Scanner!
        
        Your verification code is: {verification_code}
        
        This code is valid for 30 minutes. Please do not share it with anyone.
        
        After verification, your account will be reviewed by our admin team.
        
        If you didn't request this verification, please ignore this email.
        
        © 2025 Navvi Corporation. All rights reserved.
        """
        
        return self.send_email(to_email, subject, html_body, text_body)
    
    def send_approval_email(self, to_email: str, full_name: str, assessment_token: str, assessment_url: str) -> bool:
        """Send account approval email"""
        subject = "Your Datrix™ Account Has Been Approved!"
        
        html_body = f"""
        <!DOCTYPE html>
        <html>
        <head>
            <style>
                body {{ font-family: Arial, sans-serif; line-height: 1.6; color: #333; }}
                .container {{ max-width: 600px; margin: 0 auto; padding: 20px; }}
                .header {{ background: linear-gradient(135deg, #10b981 0%, #059669 100%); color: white; padding: 30px; text-align: center; border-radius: 10px 10px 0 0; }}
                .content {{ background: #f9f9f9; padding: 30px; border-radius: 0 0 10px 10px; }}
                .button {{ background: #10b981; color: white; padding: 15px 40px; text-decoration: none; border-radius: 5px; display: inline-block; margin: 20px 0; font-weight: bold; }}
                .footer {{ text-align: center; margin-top: 20px; color: #666; font-size: 12px; }}
            </style>
        </head>
        <body>
            <div class="container">
                <div class="header">
                    <h1>🎉 Account Approved!</h1>
                    <p>Datrix™ Business Intelligence Scanner</p>
                </div>
                <div class="content">
                    <h2>Congratulations {full_name}!</h2>
                    <p>Your Datrix™ account has been approved by our admin team.</p>
                    <p>You can now access your personalized business intelligence assessment.</p>
                    
                    <div style="text-align: center;">
                        <a href="{assessment_url}" class="button">Start Your Assessment</a>
                    </div>
                    
                    <p><strong>What's Next?</strong></p>
                    <ul>
                        <li>Complete the 27-question business assessment</li>
                        <li>Receive your detailed business intelligence report</li>
                        <li>Get personalized recommendations for improvement</li>
                    </ul>
                    
                    <p>Your assessment token: <strong>{assessment_token}</strong></p>
                    
                    <div class="footer">
                        <p>© 2025 Navvi Corporation. All rights reserved.</p>
                        <p>Need help? Contact us at support@navvicorp.com</p>
                    </div>
                </div>
            </div>
        </body>
        </html>
        """
        
        text_body = f"""
        Congratulations {full_name}!
        
        Your Datrix™ account has been approved by our admin team.
        
        You can now access your personalized business intelligence assessment.
        
        Start your assessment here: {assessment_url}
        
        Your assessment token: {assessment_token}
        
        What's Next?
        - Complete the 27-question business assessment
        - Receive your detailed business intelligence report
        - Get personalized recommendations for improvement
        
        © 2025 Navvi Corporation. All rights reserved.
        Need help? Contact us at support@navvicorp.com
        """
        
        return self.send_email(to_email, subject, html_body, text_body)


# Global email service instance
_email_service = None

def get_email_service() -> EmailService:
    """Get or create email service instance"""
    global _email_service
    if _email_service is None:
        _email_service = EmailService()
    return _email_service

File: frontend/src/services/emailService.js
/**
 * Email Service using Resend API
 * Handles all email communications for Datrix Business Intelligence Scanner
 */

const RESEND_API_KEY = import.meta.env.VITE_RESEND_API_KEY;
const RESEND_API_URL = 'https://api.resend.com/emails';
const FROM_EMAIL = 'Datrix™ <onboarding@resend.dev>'; // Use Resend's test domain for now
const ADMIN_EMAIL = 'admin@navvicorp.com';

/**
 * Send email via Resend API
 */
async function sendEmail({ to, subject, html }) {
  try {
    const response = await fetch(RESEND_API_URL, {
      method: 'POST',
      headers: {
        'Authorization': `Bearer ${RESEND_API_KEY}`,
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        from: FROM_EMAIL,
        to: Array.isArray(to) ? to : [to],
        subject,
        html,
      }),
    });

    if (!response.ok) {
      const error = await response.json();
      throw new Error(`Resend API error: ${JSON.stringify(error)}`);
    }

    const data = await response.json();
    console.log('Email sent successfully:', data);
    return { success: true, data };
  } catch (error) {
    console.error('Failed to send email:', error);
    return { success: false, error: error.message };
  }
}

/**
 * Send registration verification email to user
 */
export async function sendVerificationEmail(email, verificationCode, userName, additionalData = {}) {
  const subject = 'Welcome to Datrix™ - Verify Your Email';
  const html = `
    <!DOCTYPE html>
    <html>
    <head>
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width, initial-scale=1.0">
    </head>
    <body style="margin: 0; padding: 0; font-family: Arial, sans-serif; background-color: #f5f7fa;">
      <table width="100%" cellpadding="0" cellspacing="0" style="background-color: #f5f7fa; padding: 40px 20px;">
        <tr>
          <td align="center">
            <table width="600" cellpadding="0" cellspacing="0" style="background-color: #ffffff; border-radius: 8px; overflow: hidden; box-shadow: 0 2px 8px rgba(0,0,0,0.1);">
              <tr>
                <td style="background: linear-gradient(135deg, #3B5998 0%, #2d4373 100%); padding: 40px 30px; text-align: center;">
                  <h1 style="color: #ffffff; margin: 0; font-size: 28px; font-weight: bold;">Datrix™</h1>
                  <p style="color: #F7941D; margin: 10px 0 0 0; font-size: 14px; font-weight: 500;">Business Intelligence Scanner</p>
                </td>
              </tr>
              <tr>
                <td style="padding: 40px 30px;">
                  <h2 style="color: #333333; margin: 0 0 20px 0; font-size: 24px;">Welcome, ${userName}!</h2>
                  <p style="color: #666666; line-height: 1.6; margin: 0 0 20px 0; font-size: 16px;">
                    Thank you for registering with Datrix™ Business Intelligence Scanner by Navvi Corporations.
                  </p>
                  <p style="color: #666666; line-height: 1.6; margin: 0 0 30px 0; font-size: 16px;">
                    Your verification code is:
                  </p>
                  <table width="100%" cellpadding="0" cellspacing="0">
                    <tr>
                      <td align="center" style="padding: 0 0 30px 0;">
                        <div style="display: inline-block; background-color: #f8f9fa; border: 2px solid #3B5998; border-radius: 8px; padding: 20px 40px;">
                          <span style="font-size: 36px; font-weight: bold; color: #3B5998; letter-spacing: 8px;">${verificationCode}</span>
                        </div>
                      </td>
                    </tr>
                  </table>
                  <p style="color: #999999; line-height: 1.6; margin: 0; font-size: 14px; text-align: center;">
                    This code will expire in 10 minutes.
                  </p>
                </td>
              </tr>
              <tr>
                <td style="background-color: #f8f9fa; padding: 30px; text-align: center; border-top: 1px solid #e9ecef;">
                  <p style="color: #3B5998; margin: 0 0 10px 0; font-size: 18px; font-weight: bold;">Navvi Corporations</p>
                  <p style="color: #F7941D; margin: 0 0 15px 0; font-size: 14px; font-style: italic;">Cut Waste. Build Systems. Boost Profits.</p>
                  <p style="color: #999999; margin: 0; font-size: 12px;">
                    © ${new Date().getFullYear()} Navvi Corporations. All rights reserved.
                  </p>
                </td>
              </tr>
            </table>
          </td>
        </tr>
      </table>
    </body>
    </html>
  `;

  return sendEmail({ to: email, subject, html });
}

/**
 * Send notification to admin about new registration
 */
export async function sendAdminNotification(userData) {
  const { name, email, businessName, mobile, designation } = userData;
  const subject = `New Registration: ${name} from ${businessName}`;
  const html = `
    <!DOCTYPE html>
    <html>
    <head>
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width, initial-scale=1.0">
    </head>
    <body style="margin: 0; padding: 0; font-family: Arial, sans-serif; background-color: #f5f7fa;">
      <table width="100%" cellpadding="0" cellspacing="0" style="background-color: #f5f7fa; padding: 40px 20px;">
        <tr>
          <td align="center">
            <table width="600" cellpadding="0" cellspacing="0" style="background-color: #ffffff; border-radius: 8px; overflow: hidden;">
              <tr>
                <td style="background-color: #3B5998; padding: 30px; text-align: center;">
                  <h1 style="color: #ffffff; margin: 0; font-size: 24px;">New Registration Alert</h1>
                </td>
              </tr>
              <tr>
                <td style="padding: 40px 30px;">
                  <h2 style="color: #333333; margin: 0 0 20px 0; font-size: 20px;">Registration Details</h2>
                  <table width="100%" cellpadding="10" cellspacing="0" style="border: 1px solid #e9ecef; border-radius: 5px;">
                    <tr style="background-color: #f8f9fa;">
                      <td style="padding: 12px; font-weight: bold; color: #333333; width: 30%;">Name:</td>
                      <td style="padding: 12px; color: #666666;">${name}</td>
                    </tr>
                    <tr>
                      <td style="padding: 12px; font-weight: bold; color: #333333;">Email:</td>
                      <td style="padding: 12px; color: #666666;">${email}</td>
                    </tr>
                    <tr style="background-color: #f8f9fa;">
                      <td style="padding: 12px; font-weight: bold; color: #333333;">Company:</td>
                      <td style="padding: 12px; color: #666666;">${businessName}</td>
                    </tr>
                    <tr>
                      <td style="padding: 12px; font-weight: bold; color: #333333;">Designation:</td>
                      <td style="padding: 12px; color: #666666;">${designation || 'Not provided'}</td>
                    </tr>
                    <tr style="background-color: #f8f9fa;">
                      <td style="padding: 12px; font-weight: bold; color: #333333;">Mobile:</td>
                      <td style="padding: 12px; color: #666666;">${mobile || 'Not provided'}</td>
                    </tr>
                  </table>
                  <p style="color: #666666; line-height: 1.6; margin: 30px 0 0 0; font-size: 14px;">
                    Please review this registration in the admin panel and create an assessment invitation if approved.
                  </p>
                </td>
              </tr>
            </table>
          </td>
        </tr>
      </table>
    </body>
    </html>
  `;

  return sendEmail({ to: ADMIN_EMAIL, subject, html });
}

/**
 * Send assessment link to user after admin approval
 */
export async function sendAssessmentLink({ name, email, company, assessmentLink, expiresAt }) {
  const subject = 'Your Datrix™ Assessment Link is Ready';
  const expiryDate = new Date(expiresAt).toLocaleDateString('en-US', { 
    year: 'numeric', 
    month: 'long', 
    day: 'numeric' 
  });
  
  const html = `
    <!DOCTYPE html>
    <html>
    <head>
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width, initial-scale=1.0">
    </head>
    <body style="margin: 0; padding: 0; font-family: Arial, sans-serif; background-color: #f5f7fa;">
      <table width="100%" cellpadding="0" cellspacing="0" style="background-color: #f5f7fa; padding: 40px 20px;">
        <tr>
          <td align="center">
            <table width="600" cellpadding="0" cellspacing="0" style="background-color: #ffffff; border-radius: 8px; overflow: hidden;">
              <tr>
                <td style="background: linear-gradient(135deg, #3B5998 0%, #2d4373 100%); padding: 40px 30px; text-align: center;">
                  <h1 style="color: #ffffff; margin: 0; font-size: 28px; font-weight: bold;">Datrix™</h1>
                  <p style="color: #F7941D; margin: 10px 0 0 0; font-size: 14px; font-weight: 500;">Business Intelligence Scanner</p>
                </td>
              </tr>
              <tr>
                <td style="padding: 40px 30px;">
                  <h2 style="color: #333333; margin: 0 0 20px 0; font-size: 24px;">Hello, ${name}!</h2>
                  <p style="color: #666666; line-height: 1.6; margin: 0 0 20px 0; font-size: 16px;">
                    Your registration for <strong>${company}</strong> has been approved!
                  </p>
                  <p style="color: #666666; line-height: 1.6; margin: 0 0 30px 0; font-size: 16px;">
                    You can now access your personalized business intelligence assessment.
                  </p>
                  <table width="100%" cellpadding="0" cellspacing="0" style="background-color: #f8f9fa; border-left: 4px solid #F7941D; margin: 0 0 30px 0;">
                    <tr>
                      <td style="padding: 20px;">
                        <p style="color: #333333; margin: 0 0 10px 0; font-size: 14px; font-weight: bold;">⏱️ Estimated Time: 15-20 minutes</p>
                        <p style="color: #666666; margin: 0; font-size: 14px;">📅 Link Expires: ${expiryDate}</p>
                      </td>
                    </tr>
                  </table>
                  <table width="100%" cellpadding="0" cellspacing="0">
                    <tr>
                      <td align="center" style="padding: 0 0 30px 0;">
                        <a href="${assessmentLink}" style="display: inline-block; background-color: #F7941D; color: #ffffff; text-decoration: none; padding: 18px 50px; border-radius: 5px; font-size: 18px; font-weight: bold;">
                          Start Assessment Now
                        </a>
                      </td>
                    </tr>
                  </table>
                </td>
              </tr>
              <tr>
                <td style="background-color: #f8f9fa; padding: 30px; text-align: center; border-top: 1px solid #e9ecef;">
                  <p style="color: #3B5998; margin: 0 0 10px 0; font-size: 18px; font-weight: bold;">Navvi Corporations</p>
                  <p style="color: #F7941D; margin: 0 0 15px 0; font-size: 14px; font-style: italic;">Cut Waste. Build Systems. Boost Profits.</p>
                </td>
              </tr>
            </table>
          </td>
        </tr>
      </table>
    </body>
    </html>
  `;

  return sendEmail({ to: email, subject, html });
}

/**
 * Send assessment completion notification with report
 */
export async function sendReportEmail(email, reportUrl, userName, company) {
  const subject = 'Your Datrix™ Business Intelligence Report is Ready';
  const html = `
    <!DOCTYPE html>
    <html>
    <head>
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width, initial-scale=1.0">
    </head>
    <body style="margin: 0; padding: 0; font-family: Arial, sans-serif; background-color: #f5f7fa;">
      <table width="100%" cellpadding="0" cellspacing="0" style="background-color: #f5f7fa; padding: 40px 20px;">
        <tr>
          <td align="center">
            <table width="600" cellpadding="0" cellspacing="0" style="background-color: #ffffff; border-radius: 8px; overflow: hidden;">
              <tr>
                <td style="background: linear-gradient(135deg, #3B5998 0%, #2d4373 100%); padding: 40px 30px; text-align: center;">
                  <h1 style="color: #ffffff; margin: 0; font-size: 28px; font-weight: bold;">✅ Assessment Complete!</h1>
                  <p style="color: #F7941D; margin: 10px 0 0 0; font-size: 14px; font-weight: 500;">Your Business Intelligence Report is Ready</p>
                </td>
              </tr>
              <tr>
                <td style="padding: 40px 30px;">
                  <h2 style="color: #333333; margin: 0 0 20px 0; font-size: 24px;">Congratulations, ${userName}!</h2>
                  <p style="color: #666666; line-height: 1.6; margin: 0 0 20px 0; font-size: 16px;">
                    Thank you for completing the Datrix™ Business Intelligence Assessment for <strong>${company}</strong>.
                  </p>
                  <table width="100%" cellpadding="0" cellspacing="0">
                    <tr>
                      <td align="center" style="padding: 0 0 30px 0;">
                        <a href="${reportUrl}" style="display: inline-block; background-color: #F7941D; color: #ffffff; text-decoration: none; padding: 18px 50px; border-radius: 5px; font-size: 18px; font-weight: bold;">
                          View Your Report
                        </a>
                      </td>
                    </tr>
                  </table>
                </td>
              </tr>
              <tr>
                <td style="background-color: #f8f9fa; padding: 30px; text-align: center; border-top: 1px solid #e9ecef;">
                  <p style="color: #3B5998; margin: 0 0 10px 0; font-size: 18px; font-weight: bold;">Navvi Corporations</p>
                  <p style="color: #F7941D; margin: 0 0 15px 0; font-size: 14px; font-style: italic;">Cut Waste. Build Systems. Boost Profits.</p>
                </td>
              </tr>
            </table>
          </td>
        </tr>
      </table>
    </body>
    </html>
  `;

  return sendEmail({ to: email, subject, html });
}

